cmd/clusterctl/client/cluster/mover.go

```diff
@@ -592,6 +805,103 @@ func (o *objectMover) createTargetObject(nodeToCreate *node, toProxy Proxy) erro
 	return nil
 }
 
+func (o *objectMover) saveTargetObject(nodeToCreate *node, directory string) error {
+	log := logf.Log
+	log.V(1).Info("Saving", nodeToCreate.identity.Kind, nodeToCreate.identity.Name, "Namespace", nodeToCreate.identity.Namespace)
+
+	cFrom, err := o.fromProxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	// Get the source object
+	obj := &unstructured.Unstructured{}
+	obj.SetAPIVersion(nodeToCreate.identity.APIVersion)
+	obj.SetKind(nodeToCreate.identity.Kind)
+	objKey := client.ObjectKey{
+		Namespace: nodeToCreate.identity.Namespace,
+		Name:      nodeToCreate.identity.Name,
+	}
+
+	if err := cFrom.Get(ctx, objKey, obj); err != nil {
+		return errors.Wrapf(err, "error reading %q %s/%s",
+			obj.GroupVersionKind(), obj.GetNamespace(), obj.GetName())
+	}
+
+	// Get JSON for object and write it into the configured directory
+	byObj, err := obj.MarshalJSON()
+	if err != nil {
+		return err
+	}
+
+	filenameObj := nodeToCreate.identity.Kind + "_" + nodeToCreate.identity.Name + "_" + nodeToCreate.identity.Namespace
+	objectFile := filepath.Join(directory, filenameObj)
+
+	err = ioutil.WriteFile(objectFile, byObj, 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *objectMover) restoreTargetObject(nodeToCreate *node, toProxy Proxy) error {
+	log := logf.Log
+	log.V(1).Info("Restoring", nodeToCreate.identity.Kind, nodeToCreate.identity.Name, "Namespace", nodeToCreate.identity.Namespace)
+
+	// Get the source object
+	obj := nodeToCreate.rawObject
+	obj.SetAPIVersion(nodeToCreate.identity.APIVersion)
+	obj.SetKind(nodeToCreate.identity.Kind)
+
+	// New objects cannot have a specified resource version. Clear it out.
+	obj.SetResourceVersion("")
+
+	// Removes current OwnerReferences
+	obj.SetOwnerReferences(nil)
+
+	// Rebuild all the OwnerReferences using the newUID of the owner nodes.
+	if len(nodeToCreate.owners) > 0 {
+		ownerRefs := []metav1.OwnerReference{}
+		for ownerNode := range nodeToCreate.owners {
+			ownerRef := metav1.OwnerReference{
+				APIVersion: ownerNode.identity.APIVersion,
+				Kind:       ownerNode.identity.Kind,
+				Name:       ownerNode.identity.Name,
+				UID:        ownerNode.newUID, // Use the owner's newUID read from the target management cluster (instead of the UID read during discovery).
+			}
+
+			// Restores the attributes of the OwnerReference.
+			if attributes, ok := nodeToCreate.owners[ownerNode]; ok {
+				ownerRef.Controller = attributes.Controller
+				ownerRef.BlockOwnerDeletion = attributes.BlockOwnerDeletion
+			}
+
+			ownerRefs = append(ownerRefs, ownerRef)
+		}
+
+		obj.SetOwnerReferences(ownerRefs)
+	}
```

Comment:
This code seems a duplication of what we have in createTargetObject. What about creating a utility func and share?